### PR TITLE
Update Core Set event card implementations

### PR DESCRIPTION
**Pull request by DukeTax**
* Standardises messages.
* Converts implementations to use the cost and/or target API.
* Revert previous destructuring of event object.
* Removes redundant activePromptitle properties.
* Invokes game.transferGold() instead of two times invoking game.addGold().
* Changes var into let where possible.

### CODE REVIEW

File: server/game/cards/events/01/tearsoflys.js [hunk 1]
```diff
@@ -5,32 +5,22 @@ class TearsOfLys extends DrawCard {
...
             when: {
-                afterChallenge: ({challenge}) => (
-                    challenge.attackingPlayer === this.controller &&
-                    challenge.winner === this.controller &&
-                    challenge.challengeType === 'intrigue'
-                )
+                afterChallenge: event => event.challenge.attackingPlayer === this.controller && event.challenge.winner === this.controller &&
+                                         event.challenge.challengeType === 'intrigue'
             },
-            handler: () => {
```
Comment (cryogen) [inline server/game/cards/events/01/tearsoflys.js]: Seems like this choice was made for a reason, don't think we should be undoing that

File: server/game/cards/events/01/unbowedunbentunbroken.js [hunk 2]
```diff
@@ -41,13 +38,14 @@ class UnbowedUnbentUnbroken extends DrawCard {
...
 
-        this.game.addMessage('{0} uses {1} to make {2} unable to initiate {3} challenges until the end of the phase', player, this, otherPlayer, challengeType);
+        this.game.addMessage('{0} plays {1} to make {2} unable to initiate {3} challenges until the end of the phase', player, this, otherPlayer, challengeType);
 
         return true;
     }
 
     cancel(player) {
-        this.game.addMessage('{0} cancel the resolution of {1}', player, this);
+        //I really want to get rid of stuff like this
```
Comment (cryogen) [inline server/game/cards/events/01/unbowedunbentunbroken.js]: Things should always be cancellable and if they are then a message should be logged so people know if people are abusing it to cheat.  Also the code isn't the place to make these comments :)

File: server/game/cards/events/01/tearsoflys.js (hunk 1 above)
Comment (DukeTax) [inline server/game/cards/events/01/tearsoflys.js]: I've discussed this with @ystros not too long ago. Using the target API, even when the card technically doesn't target has two advantages: cleaner looking card implementations and not getting prompted when the card effect does not have the chance to change the game state. A disadvantage could arise when something like High Septon gets implemented. Even so, that might be resolved by adding a flag to target, something like trueTarget: false in this case.

File: server/game/cards/events/01/unbowedunbentunbroken.js (hunk 2 above)
Comment (DukeTax) [inline server/game/cards/events/01/unbowedunbentunbroken.js]: I know your stance on this, I just have a lot of issues with it. :) It's only a handful cards in the entire cardpool that are implemented like this, and adding such functionality across the board would add extra clicks, cause problematic interactions (do you want to give new players/grievers the ability to cancel Pentoshi's When Revealed?) and be a shitload of work to boot.

Fair enough about the comment. I'll record it somewhere else. :)

**Pull request merged**